integration/spark-common/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala

```diff
@@ -130,9 +135,28 @@ class CarbonMergeFilesRDD(
   extends CarbonRDD[String](ss, Nil) {
 
   override def internalGetPartitions: Array[Partition] = {
-    segments.zipWithIndex.map {s =>
-      CarbonMergeFilePartition(id, s._2, s._1)
-    }.toArray
+    if (carbonTable.isHivePartitionTable) {
+      // in case of partition table make rdd partitions per partition of the carbon table
+      val partitionPaths: java.util.Map[String, String] = new java.util.HashMap()
+      segments.foreach(segment => {
+        val partitionSpecs = SegmentFileStore.getPartitionSpecs(segment, carbonTable.getTablePath)
```

Comment:
1) SegmentFileStore.getPartitionSpecs is reading table status file for each segment. So, read & send the load details from outside to avoid this.
2) partitionLoc may be same for each segment. So, at the end only one segment per partition will remain in map. Please correct to use list based map instead

Reply:
done